src/map/packet_system.cpp

```diff
@@ -785,14 +786,25 @@ void SmallPacket0x01A(map_session_data_t* session, CCharEntity* PChar, int8* dat
         // bunch of gysahl greens
         uint8 slotID = PChar->getStorage(LOC_INVENTORY)->SearchItem(4545);
 
-        if (slotID != ERROR_SLOTID)
-        {
-            charutils::UpdateItem(PChar, LOC_INVENTORY, slotID, -1);
-
-            PChar->pushPacket(new CInventoryFinishPacket());
-            PChar->pushPacket(new CChocoboDiggingPacket(PChar));
-        }
-        else{
+		if (slotID != ERROR_SLOTID)
+		{
+			bool canDig = diggingutils::CanPlayerDig(PChar);
+
+			if (canDig)
+			{
+				charutils::UpdateItem(PChar, LOC_INVENTORY, slotID, -1);
+
+				PChar->pushPacket(new CInventoryFinishPacket());
+				PChar->pushPacket(new CChocoboDiggingPacket(PChar));
+				diggingutils::Dig(PChar);
+				Sleep(1000);
```

Comment:
absolutely not - you don't want to pause the entire server for a second every time someone digs
also, Sleep is a windows function and wouldn't compile on linux